### ISSUE TITLE
Update on the project

### ISSUE THREAD

**Issue by djaiss** [monicahq/monica#6626]
Hi. Monica's team here.

The project is not dead. Far from it. In fact we've been working for a year and a half on Chandler, the code name of the brand new major version, and we push new code almost daily.

It's available here: https://github.com/monicahq/chandler

Since Monica is still a side project and we have full time jobs and families, we can't answer all the issues that you all create, even though we read them all. We need to prioritize what we do in our limited free time. This is why we put all our energy in creating a brand new experience with Chandler.

Chandler will address a lot of problems Monica has. Not all of them, but we hope it'll make you happy. It's also heavily unit-tested so we hope there will be less bugs. The dev stack remains mainly the same (Laravel (PHP) + VueJS + Mysql/SQLite/Postgre support), but we use Vue throughout the entire app, not just in some cases like we currently have in Monica. For the curious we use https://inertiajs.com to make it possible to use Vue with Laravel. Using Vue for the front end makes a user interface that feels really fast. The UI is also cleaner and more modern, while being sober.

We plan to move Chandler's code to this current repository. Why?

- we don't want to lose these precious Github stars, because we need some sort of meaningless gratifications as developers,
- we think it's easier for the community to have one point of reference about Monica instead of several repositories. Many users point to this repo directly, we don't want to dilute your attention.

Merging two repositories into one is a hard task, especially when we don't want to lose history of both commit trees. That'll be fun to do.

Before launching Chandler officially, we need to finish an exporter/importer. Chandler's codebase and data structure is completely different from Monica's. Right now, you couldn't import Monica's data into Chandler, so that would suck. We know it's an issue.

Chandler will be more focused on documenting your life, than being a personal CRM. While we kept most of all the current features Monica has, we've put a lot of effort on the journaling/personal diary feature. I mean, a lot of effort. Honestly I think we'll have one of the best journaling system out there and we really hope you'll like it.

So, please be patient. We'll post a new issue when Chandler is ready for testing.

Below are some screenshots of Chandler, filled with fake data.

<img src="https://user-images.githubusercontent.com/61099/227721090-c0d227d9-e3db-48f2-b29e-4125df7e5803.png" width="23%"></img> <img src="https://user-images.githubusercontent.com/61099/227721093-3a1558dd-76f0-4795-afb1-1e1f636b99a6.png" width="23%"></img> <img src="https://user-images.githubusercontent.com/61099/227721095-e48b0bae-ba22-4be1-9e89-e19cdd92b04f.png" width="23%"></img> <img src="https://user-images.githubusercontent.com/61099/227721096-a9f10a62-5051-4726-bb05-8be21597ddc4.png" width="23%"></img>

**Comment by PerfectSlayer** [comment]
> Merging two repositories into one is a hard task, especially when we don't want to lose history of both commit trees. 

What's the issue? What about having two different branches without related parent so you can import Chandler branch right into this repo.

**Comment by turbopixel** [comment]
I installed and tested Chandler a few days ago. The first impression is really great and the new structure of Chandler is the right step. Monica as a personal CRM is a great tool and fun to work with, but not perfect in detail.

**Comment by Szeraax** [comment]
Have you considered creating a discord server dedicated to Monica/Chandler? That would be a helpful space for people to help each other in the community and make it so that "all discussion" about the project doesn't have to happen here in the Issues tracker. You don't even have to actively pay attention to the channels there if you don't want to, it can be focused around users helping each other. 

I really like Monica. I LOVE how you can turn off the stuff that you don't care about using it for. Really well crafted tool and I look forward to using Chandler when that branch comes here and the migration is available. Great work all!

**Comment by djaiss** [comment]
> Have you considered creating a discord server dedicated to Monica/Chandler?

Yes, but we can't keep up with all the feedback we already receive. Do you think this kind of space would be acceptable if we don't actually contribute to it? It's already hard to find time to work on the project - if we need to maintain a new community on the side, we'll never make it alive :-)

> I really like Monica. I LOVE how you can turn off the stuff that you don't care about using it for. Really well crafted tool and I look forward to using Chandler when that branch comes here and the migration is available. Great work all!

Thanks for your kind words!

**Comment by Szeraax** [comment]
> Do you think this kind of space would be acceptable if we don't actually contribute to it?

A discord server can help REDUCE the feedback that you have coming into here. This project has over 18000 stars on github! You are doing this work without any serious business goals and people are noticing and recognizing your good work. You get a discord server setup and you'll have various ambassadors hanging out there. 

For example: https://github.com/monicahq/monica/issues/6624

This is a perfect example of something where they user is not submitting a bug report or a feature request. Its a "support request". Channel support requests to discord where you have those Monica ambassadors that can help out and then just leave it alone. 

If your concern is helping to run the discord, just think about those 18000 stars on github. There are people who love Monica that would be very capable of setting up and running the discord for you (organizing channels, recognizing people who are helpful in the community with distinguishments, etc.). Heck, I'd set one up for you just to take a look at and think about if you want me to. 

Personally, I'd suggest aggressively pruning your issues and getting stricter with your minimum requirements for what comes into the issues tracker. This would help prepare for the migration where Chandler comes to take over Monica. 

What help do you feel like you need most with the project right now? Perhaps I can help (I am not a heavy programmer, so don't ask me to do big stuff in laravel).

**Comment by djaiss** [comment]
My concern is definitely about running the discord. I know @asbiin and I won't have much time to interact with the community, as much as we want to. We are just limited with our time, like everybody else. And I agree that the increasing number of issues without any answers from us is an indication of that.



> Personally, I'd suggest aggressively pruning your issues and getting stricter with your minimum requirements for what comes into the issues tracker. This would help prepare for the migration where Chandler comes to take over Monica.

I think we'll prune the issue once Chandler is out, to start from scratch.



> What help do you feel like you need most with the project right now? Perhaps I can help (I am not a heavy programmer, so don't ask me to do big stuff in laravel).

Marketing, definitely, and community management. Chandler's development is going well, new features are added at a rapid pace. But marketing? Community? This is the hard part for us geeks.

**Comment by Szeraax** [comment]
>My concern is definitely about running the discord. I know @asbiin and I won't have much time to interact with the community, as much as we want to.

Ya, that's understandable and us community members generally want you working on the parts that are useful and interesting. Not getting bogged down trying to figure out why someones armv9 CPU isn't properly handling some php instructions or whatever else. You don't need to be active for it to be a successful community resource.

> I think we'll prune the issue once Chandler is out, to start from scratch.

Do you know if github has a nice issues migration tool to move the existing issues to Monica once it is out? If not, you may want to consider moving to the Monica repo sooner than later.

> why Discord?

Discord is _very_ approachable for people to join. Many self-hosted apps use discord as their "community hub". Some prefer to use matrix chat. Both are valid, but Discord is definitely more approachable for people to join and participate in.

**Comment by Joniator** [comment]
But discord is terrible if you are a new user. It's not searchable, to fix common issues you need to have a discord account and join a server that might not interest you. You get the same 15 questions in the channels, and can't really link to the common answer (like an existing ticket).

Discord might be great for enthusiasts, that want to show off some cool hacks/tricks/strategies they use, but it absolutely sucks as support and actively turns me away if the only way to look for guidance is a discord, even if they have a FAQ-Channel

**Comment by Szeraax** [comment]
That's the hard part about community management: You can't please _everyone_. Does that mean that you should avoid doing anything? Right now, Monica has hundreds of "issues". And the developers can feel the paralysis from that number of issues that are loosely structured. 

So what should they try? If a discord helps a good portion of their userbase and reduces the number of incoming Issues (especially the ones that aren't even bugs), its their call whether to try it out. Similarly, if they do try it out and they feel like it is causing problems or failing to do what they hope, it is even easier to shut it back down. The devs have [people who WANT somewhere to hangout](https://github.com/monicahq/monica/discussions/6468#discussioncomment-5386694). Those people are the ones that can really help the project by having conversations off github that the devs never have to see because the community can answer it and scroll on by. That's what a Discord server can help with. 

Another option that the devs have is to make the Issues section dedicated to ONLY bugs. Then tags can be focused around what areas they involve or paths towards resolution. An example repo of comparable size showcases this well is[libsodium](https://github.com/jedisct1/libsodium). They only have a "bugs" template in their Issues tracker and the template says:

> THE ISSUE TRACKER IS DEDICATED TO KEEPING TRACK OF *BUGS*,
preferably after they have been already discussed and confirmed to be reproducible.
>
> FOR ASSISTANCE, PLEASE CLOSE THIS FORM AND USE THE DISCUSSIONS SECTION INSTEAD:
https://github.com/jedisct1/libsodium/discussions

This approach can be nice because github has an easy button on issues to convert them to discussions. And vice versa from discussions. The devs could also do a hybrid approach where Issues tracker is for all active work that they want to track. e.g. When they are actively looking to make a new feature, make an issue for it and plan out/discuss whatever they need to in that one issue. Similarly, when someone submits a bug, the Issues tracker would be the place to be. Thus, the issues tracker would be the "active" list of tasks and Discussions can be everything else. Using discussions doesn't preclude the devs from also having a Discord server if they want too.

@djaiss, this is all obviously up to you and what you like, so I leave it to you. You think about what you'd like to see and let me know if you want my help making it happen.

**Comment by Szeraax** [comment]
You also asked what other projects use discord. here are a few links (I'm not in all of these discord servers, I just found a couple projects in case you're curious: [calidre-web](https://github.com/janeczku/calibre-web), [Tandoor recipes](https://github.com/TandoorRecipes/recipes), and [navidrome](https://github.com/navidrome/navidrome).

**Comment by turbopixel** [comment]
Many other organizations and repositories use Discord for communication. The problem is always *time*. It needs more capacity and a permanent moderation. Not every maintainer can offer that in addition.

**Comment by Szeraax** [comment]
At least on Tandoor-recipes' discord, the project maintainers show up there when they want to. The community ambassadors (people like me who know at least a bit) hang out there and take care of many of the questions that come in. That project only has about 4k stars on github, but the discord has about 1000 members.

**Comment by nosbig** [comment]
If the concern is support issues versus development issues, another accessible option might be Answer (https://answer.dev).  It's a self-hosted version of something resembling Stack Overflow.  Has built-in moderation operations, tags and categories, and more.  It's still an early project; I will have to check to see if it has Github login integration.  That might be the easiest route to redirecting people to a place for support-type issues in a way that's searchable and open.

**Comment by djaiss** [comment]
Thanks for this discussion. Just came back from a week of vacations that I really needed.

I wonder if Discord is ideal. Wouldn't Reddit be a better place? More visibility, seems a little bit less closed than Discord in the sense that posts can be public and people can help each other there?

**Comment by Szeraax** [comment]
> Wouldn't Reddit be a better place?

I've been thinking about this discussion the last few days and no, I don't think that adding a reddit area would likely be helpful. Having 2 forum type areas just leads to increased noise if the users of both forums have large overlap. E.g. Most technical reddit users are very fine with using and logging into github. If you have a sub-reddit, it will be a ghosttown that has duplicate posts between it and here. Example: https://www.reddit.com/r/fogproject/ (contrasted to the main forum: https://forums.fogproject.org/). 

If you're going to do anything with Discord, its going to be to do something different than your Monica issues tracker and discussions area. It would be to have an area for the part of your community that likes to be able to chat together. My theory for the downvotes that my comments have received is from people who don't want to have to go to Discord to get user support. Or don't want to be forced to use Discord. Which, frankly, seems reasonable and worth listening to. 

If you want to do a better job on your issues tracker, define what you'd like to see and either start enforcing it OR empower people to help enforce it. You don't have to do it all alone. If you want to make issues only be for bugs and active development, then you'd want to create a Technical Help discussion category. And actively move items from the issues tracker to the Technical Help discussion category. 

Of course, this is all kind of interesting because you're planning to clear out all your issues at some point when you migrate Chandler into Monica. Maybe your best bet to play around with all this stuff in the Chandler repo. Maybe you should seriously consider moving Chandler into Monica repo sooner than later and do the cleanup on issues. 

Entirely up to you. You let the community know what you want done and how we can help. 

Separate from the existing github repo that you have, you can also engage another segment of your community via Discord. I still believe that doing so would help with the incoming issues to gh, but I'm fine to leave that as a disputed point if people don't agree that it would help with your incoming work.

**Comment by disconn3ct** [comment]
Discord makes it nearly impossible to reference previous answers. Instead of a single location dedicated to a topic, Discord history is a time-based events list with all active topics interleaved and smeared out across other, unrelated discussions.

Are your issue-notification emails overwhelming now? Try reading them in chronological order instead, unthreaded. This is the direct user experience trying to find existing data on Discord.

This isn't just annoying; it has the direct effect of preventing useful searches before asking, which leads to lots of negative interactions in the community.

I would second the suggestion that this is a good time to separate issues, feature requests and discussions (including support requests and everything else). Then you can point stalebot at the feature requests tag and let the community handle the discussions.

**Comment by phorcys420** [comment]
I think Discord is a convenient tool that almost everyone can pick up quite easily and that a lot of people are already active on.

This makes it "better" than slack for people like me which already have a Discord account, for example, I have a Slack account for a single project, which makes me not want to use slack nor ask help in that because I don't to bother with it.

I think Matrix is also a safe bet but not many people use it, maybe the best solution would be to have a Matrix <-> Discord bridge in place. I would be happy to set it up if needed.

[Coder](https://coder.com) uses a Discord for assisting users and feedback.

I think it works pretty well especially with the "new" forum channels feature.
We really have managed to make it a space where it's not hard to search back for older issues because of the current setup.

I am the one administering the Codercord bot and I would be glad to port it if a Monica Discord ever opened.

Feel free to check it out at [discord.gg/coder](https://discord.gg/coder).
You can create threads titled "Monica test thread" in #help and I'll delete them.
I think it's worth checking out because the system we have in place with that bot is liked by our users and I think that it's honestly really good to have.

**Comment by opie4624** [comment]
Discord now forces people to pay to be members of more servers, which means now I have to delete some of the projects I'm part of if I want to join another.

**Comment by phorcys420** [comment]
> Discord now forces people to pay to be members of more servers, which means now I have to delete some of the projects I'm part of if I want to join another.

To clarify, "regular" discord users can join up to 100 servers/guilds, nitro members can join up to 200, i don't know if it's that big of a deal for most people

**Comment by benjick** [comment]
Maybe the noisy discord discussion should be in another issue since this issue is about chandler.

Any word on when we can expect chandler to be merged into main and officially released? I'm excited

**Comment by djaiss** [comment]
Hope you will like it.

Also, sorry for all the bugs we've seen in production, folks. We are actively looking into them.

**Comment by benjick** [comment]
Are there any self-hosting examples for Chandler? I tried looking for docker-compose files in the `chandler` branch but I couldn't find any

**Comment by stanleyshly** [comment]
Same, Chandler in the original repo had an option to use docker to host, as well as instructions on how to do so but I don't find anything in the `chandler` branch.

**Comment by pl33x** [comment]
Would also like a hint how to setup chandler especially with sqlite via docker-compose

**Comment by despokd** [comment]
> Would also like a hint how to setup chandler especially with sqlite via docker-compose @pl33x 

Just insert the GitHub docker image `ghcr.io/monicahq/monica-next:chandler` from https://github.com/monicahq/chandler/pkgs/container/monica-next and rest is like 4.x monica

```YAML
version: "3.4"

services:
  app:
    image: ghcr.io/monicahq/monica-next:chandler
    depends_on:
      - db
    ports:
      - 8080:80
    environment:
      - APP_KEY=[insert your generated key]
      - DB_HOST=db
      - DB_USERNAME=monica
      - DB_PASSWORD=secret
    volumes:
      - data:/var/www/html/storage
    restart: always

  db:
    image: mysql:5.7
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=true
      - MYSQL_DATABASE=monica
      - MYSQL_USER=monica
      - MYSQL_PASSWORD=secret
    volumes:
      - mysql:/var/lib/mysql
    restart: always

volumes:
  data:
    name: data
  mysql:
    name: mysql
```

**Comment by phorcys420** [comment]
> Would also like a hint how to setup chandler especially with sqlite via docker-compose

Here's how I do it with MySQL, you should be able to get rid of the ``db`` container and use some env variable to tell it to use SQLite instead.

```yml
version: "3.9"
name: "chandler"

networks:
  default:
    external: false
    ipam: 
      driver: default
      config:
        - subnet: 172.34.0.0/24

volumes:
  db:
  data:

services:
  db:
    container_name: monica_db
    image: mysql:5.7
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=true
      - MYSQL_DATABASE=monica
      - MYSQL_USER=${DB_USER}
      - MYSQL_PASSWORD=${DB_PASS}
    volumes:
      - db:/var/lib/mysql
    networks:
      default:
        ipv4_address: 172.34.0.3
    restart: unless-stopped

  monica:
    container_name: monica
    image: ghcr.io/monicahq/monica-next:chandler
    environment:
      - DB_HOST=db
      - DB_USERNAME=${DB_USER}
      - DB_PASSWORD=${DB_PASS}
      - APP_ENV=production
      - APP_URL=${MY_ACCESS_URL}
      - APP_TRUSTED_PROXIES=*
      - MAIL_MAILER=log
    volumes:
      - data:/var/www/html/storage
    networks:
      default:
        ipv4_address: 172.34.0.2
    links:
      - db:db
    restart: unless-stopped
  ```

**Comment by imawaki** [comment]
Hello there 👋 
What is the recommended way to contribute to localization?
I think I found two possible ways to do this;

* just PR on `lang/ja/*.php` or `lang/ja.json`
* via [https://crowdin.com/project/monicahq/ja](https://crowdin.com/project/monicahq/ja)

I've done some translations on crowdin.
Many thanks in advance ❤️

**Comment by phorcys420** [comment]
@djaiss even though monicahq/chandler was closed, will the issues in it be addressed?

thank you

**Comment by slifty** [comment]
Is there an FPM version of the chandler image?  I'd love to be able to just swap it into my existing docker-compose file that is built on 4.0's fpm image.

**Comment by vnghia** [comment]
For anyone who is self-hosting monica. Does your `Available modules:` have some modules or it's empty like me ? This makes the self-hosted version unsuable :(
![image](https://github.com/monicahq/monica/assets/17511625/bc92f033-3336-41b3-ba01-5179c429657f)

**Comment by JoshuaBehrens** [comment]
@vnghia are you sure this is monica? Is this already chandler? I am hosting monica 3 myself but this is not looking like yours.

**Comment by vnghia** [comment]
Ah sorry, it is about chandler. Basically I can get it up and running but it is not usable and I can not create any modules or so. I would like to ask how is your experience with self hosted chandler so far ?

**Comment by stephenwade** [comment]
@vnghia I had the same experience. No templates or modules by default.

**Comment by despokd** [comment]
> Ah sorry, it is about chandler. Basically I can get it up and running but it is not usable and I can not create any modules or so. I would like to ask how is your experience with self hosted chandler so far ?

Could be a not missing database connection. Enabling debug mode can help. At start I had "missing migrations" messages until I correctly setup the database.

**Comment by vnghia** [comment]
All my migrations are done and I can add new contact. But I can not configure the template to show information about these contacts and I only have their name for now :/ One more thing I found out missing is the list of currencies which is also empty. My current database is `Postgres` if that helps
![image](https://github.com/monicahq/monica/assets/17511625/40094485-0e11-41c9-897e-408d705a6afb)

**Comment by HarvsG** [comment]
> > Would also like a hint how to setup chandler especially with sqlite via docker-compose @pl33x
> 
> Just insert the GitHub docker image `ghcr.io/monicahq/monica-next:chandler` from https://github.com/monicahq/chandler/pkgs/container/monica-next and rest is like 4.x monica
> 
> ```yaml
> version: "3.4"
> 
> services:
>   app:
>     image: ghcr.io/monicahq/monica-next:chandler
>     depends_on:
>       - db
>     ports:
>       - 8080:80
>     environment:
>     # see https://github.com/monicahq/monica/blob/chandler/.env.example  
>       - APP_ENV=production
>       - APP_DEBUG=false
>       - APP_KEY=secret
>       - DB_HOST=db
>       - DB_USERNAME=monica
>       - DB_PASSWORD=secret
>       - MAIL_MAILER=smtp
>       - MAIL_HOST=smtp.server.com
>       - MAIL_PORT=465
>       - MAIL_USERNAME=email@username.com
>       - MAIL_PASSWORD=secret
>       - MAIL_ENCRYPTION=tls
>     volumes:
>       - data_chandler:/var/www/html/storage
>     restart: always
> 
>   db:
>     image: mysql:5.7
>     environment:
>       - MYSQL_RANDOM_ROOT_PASSWORD=true
>       - MYSQL_DATABASE=monica
>       - MYSQL_USER=monica
>       - MYSQL_PASSWORD=secret
>     volumes:
>       - mysql_chandler:/var/lib/mysql
>     restart: always
> 
> volumes:
>   data_chandler:
>     name: data_chandler
>   mysql_chandler:
>     name: mysql_chandler
> ```

Anyone know how to get this running on a raspberry pi (armv7?)

**Comment by samip5** [comment]
> Anyone know how to get this running on a raspberry pi (armv7?)

Update to 64-bit OS. Rapsberry Pi 3 B and everything after is 64bit capable.

**Comment by martynvdijke** [comment]
It took me a while to get the database connected correctly, I had to set these env variables to store everything in mysql instead of sqllite database.

```bash
DB_CONNECTION=mysql
DB_HOST=monica-db
DB_DATABASE=monica
DB_PORT=3306
DB_USERNAME=monica
DB_PASSWORD=secret
```

**Comment by stanleyshly** [comment]
Does anyone have a working docker-compose.yml and .env file combination? I tried replacing the image url of [ghcr.io/monicahq/monica-next:chandler](http://ghcr.io/monicahq/monica-next:chandler) with [ghcr.io/monicahq/monica-next:main](http://ghcr.io/monicahq/monica-next:main), but I’m still getting “server error” issues when trying to access the url.

**Comment by Boostflow** [comment]
Is it still possible to work with the 5.x version of Monica? Whenever I run the docker compose with the url of [ghcr.io/monicahq/monica-next:chandler](http://ghcr.io/monicahq/monica-next:chandler) it replaces with [ghcr.io/monicahq/monica-next:main](http://ghcr.io/monicahq/monica-next:main) which is the 4.x or stable release. Love to try the Chandler version in docker but I'd love to learn where to find the image?

**Comment by slifty** [comment]
I'm getting increasingly anxious about the future of this project :(

Does anybody have a line to the core maintainers? It would be great to know if they need the community to step up and help out a bit more.

**Comment by djaiss** [comment]
Hi all.

Original author here. My comment below is not an excuse, it's an attempt to explain my thoughts as honestly as possible, and it won't be super structured.

I know we've been silent on this project for a while. It's incredibly taxing to maintain a project that huge, especially considering the fact that we have a job, @asbiin and I. @asbiin has been an amazing partner on this project, and I'm thankful he's there, and if the project is where it's at right now, it's thank to him.

I think, personally, I got burnt out and got through terrible mental health issues related to open source. 2023 has been a very intense year. Much more responsibility at my job, I got married (🎉), I got fatter because I don't take time to do much sport. I got caught up in a vicious cycle: every time I checked my Github notifications, I realized how much attention this project needed, and it's stressed me tremendously, so I closed the tab (on Firefox, because Firefox > Chrome, obviously 😅). Rinse and repeat. And this is where we are now: a project that feels abandoned despite having a great community.

After a few months of thinking, on the v5 redesign, I think I got carried away of wanting to do too much. All the customization possibilities, all the possible use cases, a much larger scope than the first version of Monica. The software is now much bigger than what it needed to be originally. I'm still proud of what we've done with v5, I think it has a tremendous potential, even though we still have a lot of bugs. But by trying to be an all-in-one CRM, nothing is "wow" because everything is half baked. It's not ideal. I now would prefer a much smaller software that does a few things very well instead.

Also, one more reason about why I got burned out. The need to migrate data from the Monica to Chandler. It would take so much time to implement this, for basically a one-time action. At the current rate of what we can do, this is at least a three months job, at the minimum. Just thinking about this feels like the worst part of a daily job. Something that do not bring any joy. My current feeling is that I should let the current state of Monica v4 as it is, make it read-only, sunset the product on our hosted instance, and simply launch v5 (aka Chandler) as if v4 doesn't exist, and therefore, don't migrate data. We have so many users on our hosted instance, but since we do not have analytics, I'm not actually sure on how many active users we have. I don't know exactly how they will like this though.

All this to say that the project has become way too big for only two maintainers with full time jobs "on the side". If we were full time, I think it would be okay, but that's not the case.

Basically, we need help. On several fronts:

- on the repo itself: issues, documentation.
- on the code: bug fixing, features and API.

If I didn't want to ask for help before, it's because I've seen a lot of people motivated by the project, wanted to do something, and after a few days, disappeared.

So what must be done?

**Repository maintenance**

1. We need strict guidelines if we want more people to help the project. The clearer the documentation how the goals of the project and what we accept and what we don't accept, the better. Monica v4 has had way too many PRs that didn't bring the software to a better place.
1. 662 open issues is a nightmare. Most of the things in this tab will not be done. First of all, we should close all the issues about Monica v4. All our efforts should be put on Chandler. I don't even anymore know what needs to be done on Chandler.
    - We need to tag all the issues with the tag `v1` (https://github.com/monicahq/monica/labels/v1) so we know we can close it
1. We need documentation on features.
    - the [official documentation on Gitbook](https://docs.monicahq.com/), powered by our repo [doc-gitbook](https://github.com/monicahq/docs-gitbook)
1. We need some kind of help on the marketing part to help spread the good word on how this tool can help people.
    - we need help on the marketing site
    - we need help on mastodon/twitter and co

**On the code**

1. We need at least good bug reports, and ideally people fixing those bugs
1. We need people who will review PRs (based on strict guidelines that we would write together) and help merge PRs faster. The current rate and speed of acceptance of PRs is unacceptable at the moment, and I'm ashamed of it.
1. We need help on everything related to self host. That means tackling the bugs on Docker & co and have a great image.
1. We need to work on storing files on S3 and not on Uploadcare. We can't have dependencies like that on our open source project. Uploadcare was meant as a way to build the software faster but it sucks for self hosting.
   - The problem with that one: we also need to add a dependency on GD or ImageMagick since we need to resize avatars and photos. I really, really don't like that, but we have no choice.
   - I do not know enough to do this myself.
1. We need an API with a great documentation.

Believe it or not, we care deeply about Monica. We've put our souls into it for many many years. We are at a stage where we need your help. Without the help of the community, the project has a 100% chance to fail in the near future.

Help us build the future.

**Comment by Szeraax** [comment]
Congratz on getting married, @djaiss! We all are so happy for you. We love this project and we are very thankful for the work that you both have put into it. 

I'll reference my comment from a [year ago](https://github.com/monicahq/monica/issues/6626#issuecomment-1509261606):

> Personally, I'd suggest aggressively pruning your issues and getting stricter with your minimum requirements for what comes into the issues tracker. This would help prepare for the migration where Chandler comes to take over Monica.

257 of your 662 open issues (~30%) are feature requests. ~50 of your 662 are bugs. The other 300 items have some other label, or no label at all. 

When I think of community management and your ideal community managers, I think of deputies who HELP you out. Who make things more manageable for you. Who help to enforce standards and organization. Now _what_ they are doing, is up to you. 

You said that you want your issues to be tagged as <v5 and v5+ so that you can close everything that is <5.0.0. And you say that you want to identify what is in-scope for a viable 5.0.0 release. That's where you need to be using a project. You've already done similar for v3 I can see. Get your community managers to set that stuff up and you can guide and refine. Stuff like this is CLEAR and easy for your community managers to follow.

But issues starts to get sticky when community managers don't really know what you want to see. 

You have to pick whether you want feature requests in here https://github.com/monicahq/monica/discussions/categories/ideas or just as a label in your issues. There are pros and cons to either one, but using both seems less than ideal. Using discussions is nice because people can upvote ideas and you can sort by most upvoted ideas to see what features the community really wants. But ideas don't really get indexed by google like issues do. So if you want people to be able to find an idea by google, you need to use issues instead. 

If this were my project and I wanted to switch to using discussion ideas, I'd seriously consider having someone [convert all feature requests](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#converting-an-issue-to-a-discussion) to your "Ideas" discussion category. Again, the ideas category is nice because people can thumbs up the idea and you as a developer can sort by "most thumbs ups" at the category view. I'd seriously consider tagging the people who have previously thumbs upped (upvoted) the feature request so they know to come to the discussion idea and now thumbs up the idea. And if you want to maintain your own list of ideas that you are considering adding but want them separate from adding each one individually in the ideas category, maintain a roadmap document with those extra items as wishlist features you would like to implement. But your default idea place should be the ideas category so that people can individually upvote ideas and you can be empowered with the community feedback. 

Talking more about the cons of this approach, the primary one I see is that they aren't well indexed in google. Searching "monicahq Immich integration" has the top result of: https://github.com/monicahq/monica/discussions/categories/ideas. Why doesn't it take you straight into #7116 ? Why doesn't google have that indexed? You have to use the search tool within github to find it. Do you like being able to sort by top ideas more than you like it for people to find individual ideas easily? If you have a community manager, they could use labels to review every incoming idea and search for something similar to direct the user to. You could even have every idea also have a github issue item as well. That's quite a bit more work, but maybe its what you'd really like to see happen. I dunno.

Now, these are just some of my quick thoughts. I love this project. I'll happily volunteer to be a community manager that works to make github more useful TO YOU. If we get you setup with a project roadmap for 5.0.0 and you like the list of features, then all you have to do is hit those items and then you are done and can officially release 5.0.0. Simple as pi :D. There are others who would also be willing to be community managers besides me, by the way. But as far as I'm aware, you haven't ever put out a call to ask who is willing to take on these responsibilities. There is a certain level of trust since they can screw things up, so don't just make _everyone_ a maintainer, obviously. 

Lastly for now, I think you need to be willing to setup some extra help for your community managers. Whether its typing up a doc on specifics that you want them doing, or getting a voice call with them sometimes, or setting up some chat channel that you can use to communicate. They need help knowing when they are doing stuff right or wrong (and changes are needed). A little feedback from you towards the community managers will enable them to really help you out.

**Comment by slifty** [comment]
So much love to you @djaiss!  Please know that we're here for you, and that there is no shame in the amount of tasks there are to work on (including PRs to review).  I know exactly how you feel.

I would be very happy to be involved in any of the following:

1) issue pruning and PR review
2) community management  and communication (I'd be thrilled to work on a team with folks like @Szeraax to facilitate this, and could imagine there being value in hopping on a video call / even setting up a regular community call.  Maintaining a project board and communicating roadmap / priorities based on conversations would be great.
3) I will be glad to commit code over time as well of course, as there are features I'd like to help build, but the first two feel most important in the short term.

FIWW: I do agree that it's reasonable for you to focus on v5 and call v4 static for now.  The community can help build an importer if that is valued.

**Comment by millette** [comment]
@djaiss Merci beaucoup d'avoir fait le point. Prend d'abord soin de toi, je sais tellement à quel point l'open source peut consumer une vie (et je suis pro logiciel libre/open source depuis 30 ans).

Je pourrais écrire quelques scripts pour fermer des issues en batch. Quand à la migration, peut-être qu'un third party pourrait s'en charger? À part votre hosted monica, as-tu une idée comment c'est populaire, ça touchera combien de gens ce reset?

**Comment by djaiss** [comment]
Thanks for all your feedback! @asbiin and I appreciate it a lot.

We'll discuss he and I over the next few days to check how we will proceed to onboard the ones who want to help us. I think we will prefer a way of asynchronous communication instead of calls - we already have too many Teams calls all day long 😅

I'm investigating the route of teams within the organization but I think Github doesn't like team communication since the UX is atrocious there. We had a Slack in the past but I'm afraid of the noise there. Any ideas on how we could communicate in this future maintainers group?

**Comment by djaiss** [comment]
We should write in English here so everyone understands 😅

> Je pourrais écrire quelques scripts pour fermer des issues en batch. Quand à la migration, peut-être qu'un third party pourrait s'en charger? À part votre hosted monica, as-tu une idée comment c'est populaire, ça touchera combien de gens ce reset?

Last time I checked we had ten of thousands of self hosted instances out there. But I don't think it'll be an issue for those people. I think it's worse for us to maintain two hosted versions: the current one with 45k+ users, and the new one with already 2500+ users.

**Comment by Szeraax** [comment]
> Any ideas on how we could communicate in this future maintainers group?

We can use a discussion, honestly. You already have that space enabled in the repo, so its not like you have to setup anything fancy. Or slack/discord/email.

**Comment by slifty** [comment]
Totally makes sense regarding the preference for async.

Some thoughts, in no particular order, regarding communication:

1. Some FOSS projects I work on use [Zulip](https://zulip.com/). It is, itself, open source, and open source projects can have a free standard plan.  It's a slightly different communication model than slack / discord as you have a "stream" of conversation (similar to channels) that can contain different sub-"topics" within a given stream (similar to threads).

I don't know if this tool would address the noise concerns, but one thing it definitely has that's better than Discord / Slack is that streams can be set up for "global read" which mean conversations can be linked to / read by people without an account.  This is a nice way to avoid Discord Knowledge Rot, and I've found it to be a bit more suited for FOSS.

2. Could you create a private github repository that exists *solely* to host internal discussions?  I think you'd then have more granular control over who participates.

3. You could create a new section in the existing discussions section for the purposes of maintainer coordination to see how it goes; if noise becomes an issue there then you could look into other options with more granular access controls.

4. Email is fine too!

**Comment by phorcys420** [comment]
Hey all, first of all thanks for the heads up @djaiss.

I'd be glad to help where I can but we should probably take this to a more immediate type of discussion (e.g Discord, Slack, etc as already mentioned by others) to organize better.

Regarding the platform to use, I don't know if Zulip is a great choice, last time I used it I really hated the UX, maybe it changed though.

I think Discord is fine since a lot of FOSS communities are already on it but I think the best thing to do would be to set up a poll somewhere so we can decide on what to use.

**Comment by sylvainmetayer** [comment]
Thanks for the heads up @djaiss  !

I'd love to help as I'm an active user on my self hosted v4 instance, and would love to run my own v5 instance.

To avoid multiplication of communication channel, I'd rather stay with a Github team inside the MonicaHQ organization and an asynchronous communication (also it would be easier to link issues/PR from my perspective), but if the majority of the community move somewhere else, I'd join the majority.

When this decision is made, I'd like to help to create bug report/testing the v5 and help through the self-hosted part.

**Comment by whysthatso** [comment]
just a comment regarding communication: use discussions, it's like a self building documentation, and a good place to move issues to that are not issues.

**Comment by MyNameisStitch** [comment]
@djaiss first off, thank you to both you and @asbiin for all of your hard work and your vision to create Monica.  I'm an Aspie myself and this is the tool I've needed for so long and I want to help in any way I can.  I'm not a coder (I only know enough to be dangerous lol) but I can help with beta testing, brainstorming ideas/processes, sorting through issues, anything I can do to help I'll do my best.  I am self-hosting v4 currently.

I also agree with @whysthatso about using GitHub discussions.  I frequently turn to the issues and discussions tabs when I'm troubleshooting issues.

**Comment by Xyz00777** [comment]
Hi im a little bit confused, is chandler now life? wouldnt it be than a good idea to close these issue? or is there an point in still letting it open which i dont see?

**Comment by nabby27** [comment]
Hello @djaiss, after reading this [comment](https://github.com/monicahq/monica/issues/6626#issuecomment-2002484416) I am very sorry for what happened to you, I do not have such a large open source project but I can understand everything you say. It's a shame that despite having a project as cool as this and such an incredible community, you can't dedicate yourself 100% to the project. I recently launched a project called [Opire](https://opire.dev) that makes it easy to create rewards to solve issues, the good thing about Opire is that anyone can create rewards (not just you) in such a way that the community can economically help evolve the software. This way we can encourage other programmers to want to solve these issues. There are more advantages of using Opire that you can see in the [documentation](https://docs.opire.dev) if you are interested. I know this isn't going to solve all the problems but at least I hope it can help a little. For now, I would like to put $100 through Opire in the issue that you consider best.

**Comment by RubenRuCh** [comment]
> Hello @djaiss, after reading this [comment](https://github.com/monicahq/monica/issues/6626#issuecomment-2002484416) I am very sorry for what happened to you, I do not have such a large open source project but I can understand everything you say. It's a shame that despite having a project as cool as this and such an incredible community, you can't dedicate yourself 100% to the project. I recently launched a project called [Opire](https://opire.dev) that makes it easy to create rewards to solve issues, the good thing about Opire is that anyone can create rewards (not just you) in such a way that the community can economically help evolve the software. This way we can encourage other programmers to want to solve these issues. There are more advantages of using Opire that you can see in the [documentation](https://docs.opire.dev) if you are interested. I know this isn't going to solve all the problems but at least I hope it can help a little. For now, I would like to put $100 through Opire in the issue that you consider best.

Opire's co-founder here, and also an Aspie. I didn't know about this project before, but I can totally understand how it has been able to help other aspies and introverts folks. Discovering it has been great - I will totally try it out. I just had a very bad experience not remembering a neighbor's name and I had to deal with anxiety for a week cause of him getting mad about it 😅 I wish I had discovered Monica sooner.

I would like to contribute to Monica. As a developer (with some experience in PHP & Vue) I could help with the code, but realistically I already have too much in my plate, and you already have bad experience with people motivated about helping that end up disappearing. I could tackle some bugs and help with small maintenance issues, but definitely I can't compromise to be a proper maintainer - I just don't have enough time.

However, I'm willing to help by creating rewards myself in issues that you / the community consider as priority. Especially looking forward for those issues that will help to reduce the amount of attention that the project requires (bug fixes, maybe?). It's not a lot, but I would be able to put around ~$100 per month into funding rewards to encourage others devs to solve the issues.

I saw you have a Patreon. If you don't want to install Opire, that's fine - I'll set up a monthly donation to your Patreon. But I would prefer to fund specific issues to increase the amount of devs collaborating and reduce the pressure on you.

**Comment by CastlegateHOA** [comment]
After this week's report of a year's long takeover attempt that was thwarted by a microsoft employee on xz utils, everyone in the open source world is probably scared of anything that looks like coordinated pressure on open source _anything_. Monetarily, socially, etc. If you don't know what I'm talking about, go read this horrifying story: https://www.theverge.com/2024/4/2/24119342/xz-utils-linux-backdoor-attempt

Not trying to knock on opire specifically (never heard of it), but I'm just leery of external influence, especially recently.

**Comment by Szeraax** [comment]
After this week's report of a year's long takeover attempt that was thwarted by a microsoft employee on xz utils, everyone in the open source world is probably scared of anything that looks like coordinated pressure on open source anything. Monetarily, socially, etc. If you don't know what I'm talking about, go read this horrifying story: https://www.theverge.com/2024/4/2/24119342/xz-utils-linux-backdoor-attempt

Not trying to knock on opire specifically (never heard of it), but I'm just leery of external influence, especially recently.

**Comment by RubenRuCh** [comment]
Totally understandable @Szeraax!

However, tbh I'm glad the backdoor was tried through open-source  - at least that way people like Andres Freund (the Microsoft employee) can check the source. Imagine if this was done in private code - the person/group behind the backdoor could have perfectly infiltrated an employee in the company to make the changes without going unnoticed, and then if someone realized about the 500ms lag they couldn't have investigated what caused it

Anyway, that's just my personal opinion - I still can understand everyone being scared

**Comment by slifty** [comment]
@djaiss @Szeraax I'm not sure if y'all started coordinating off line but in the event that it would be helpful to coordinate anything outside of this thread (in lieu of setting up discussions), my email is `slifty@gmail.com`.

I'd be glad to work with @Szeraax to help implement the setup of some discussion categories; I think the first step would be just taking a moment to level set on what your expectations would be @djaiss and how we can help things feel less overwhelming on your end.

**Comment by nabby27** [comment]
> After this week's report of a year's long takeover attempt that was thwarted by a microsoft employee on xz utils, everyone in the open source world is probably scared of anything that looks like coordinated pressure on open source anything. Monetarily, socially, etc. If you don't know what I'm talking about, go read this horrifying story: https://www.theverge.com/2024/4/2/24119342/xz-utils-linux-backdoor-attempt
> 
> Not trying to knock on opire specifically (never heard of it), but I'm just leery of external influence, especially recently.

Hi @Szeraax  I totally agree, I understand the concern and distrust (especially in a project that we have just launched) but to be fair we do not make changes directly to the code (nor do we want to influence what changes are made to the code) or even we do nothing that doesn't already exist, there are competitors like Algora.io or previously bountysource that do something similar.

What my colleague and I said is completely true, we want to help the project not only monetarily but also encourage issues to be solved since it's one of the problems that the maintainer mentioned and we simply believe that this tool can help and is the best for it.

Anyway, it was just a suggestion 😄

**Comment by Szeraax** [comment]
I have no authority here, so please don't think that my voice matters any more than yours does. What happens is up to the developers. 

Personally, I'm more interested in the comment by Slifty about updates on community management and hearing how the developers think the community can help them with the community management side of this awesome project.

**Comment by slifty** [comment]
@djaiss no stress, but wanted to ping here again in case you had interest in moving forward on exploring what more community involvement could look like.

**Comment by Xav-v** [comment]
Same here, I'm ready to give some support, so let me know if any updates

**Comment by naiba** [comment]
@djaiss I read the whole post, I have an open source project with 6k stars myself, I would love to add lunar calendar support in monica v5 in an extensible way, which will also support Buddhist calendar users, my suggestion is that we have an instant chat group to gather developers who are capable and want to contribute to chat. Although I closed the PR after adding lunar calendar support in v4 because I didn't get a response for a long time, I still pay attention to this project in the past few years.

**Comment by tmuras** [comment]
Hi All,

I am big fan, user and Patreon supporter since 2018. Thank you @djaiss for all the work you've done here. It is much appreciated. 

Looks like the project needs some help now. If you are interested in helping (whatever it means) and **you have some spare time and availability to work on the project mid- and long-term** then please drop me an email to my Google account : nexor1984 . I will organize a call and mailing list where we can decide how to best help Mazarin and the project.

**Comment by hycday** [comment]
> Is it still possible to work with the 5.x version of Monica? Whenever I run the docker compose with the url of [ghcr.io/monicahq/monica-next:chandler](http://ghcr.io/monicahq/monica-next:chandler) it "manifest unknown" errors out and when using [ghcr.io/monicahq/monica-next:main](http://ghcr.io/monicahq/monica-next:main) it deploys the 4.x or stable release. Love to try the Chandler version in docker but I'd love to learn where to find the image?

tried with this one, or the other link you gave (https://hub.docker.com/r/k714040/monica , which gives me an error)
and in all cases, i get to install version 4.xx not 5.x :( 

therefore i'm really unsure how to get it to work and would love to, so if anyone has some idea please share

**Comment by Szeraax** [comment]
@hycday my beta Chandler was working last i checked. You should post a new bug report so that you have a dedicated thread for getting support. Be sure to include your docker compose file so we can review. Feel free to tag me when you post too and I'll be sure to take a look

**Comment by Szeraax** [comment]
@djaiss Hycday's comment here is a perfect example of what community managers can help with. I went ahead and told them to go file in an appropriate place so that they can get support without clogging up this channel with off-topic discussion. Because I'm not a community manager, I will admit that I went outside of my scope of authority as a fellow consumer by telling Hyc not to clog up this channel. 

Hyc opened #7333 and since it was a simple question, I was able to provide and answer and I bet the issue will be solved soon. If you have some community managers, they can help keep communication organized where you want it, even if you're wanting something like "All beta issues should go in this thread until further notice!"

**Comment by hycday** [comment]
i can confirm :) the solution worked (one minor open point which i cannot find in the documentation but i made a followup in the ticket) but besides that, it helped me (thanks again @Szeraax )

i'm hoping for some import/export features so that i can feel more at ease with a beta solution, but otherwise, quite happy with what im seeing in the demo, it's really amazing ! you deserve some funding as this is by far the best solution for a rolodex (and actually, seems like the only one) and it can be self-hosted... amazing

**Comment by VickyRampin** [comment]
I am wondering if this issue should be closed, since Chandler has been read-only for about a year at this point and this issue is pinned and directing people there?

**Comment by jskherman** [comment]
@VickyRampin @djaiss I'm a little bit confused. There's this issue, https://github.com/monicahq/chandler/issues/558, that says that it has moved to the `chandler` branch for this repo but I only see two `gh-readonly-queue/chandler/` branches and no `chandler` exactly. Is `5.x` supposed to be the `chandler` branch?

**Comment by tmuras** [comment]
I have the same feelings about the project @Pamalosebi . I'm trying to gather some people interested in continuing the work but so far there are only 3 of us. If you'd like to join, drop me an email at nexor1984 in google domain.

**Comment by stanleyshly** [comment]
Hi all. Just to check, when you say the project is dying, can you clarify what you mean? I'm currently using the beta docker version, the last version was published 2 months. For those of us who don't follow development as closely, what is happening there?

**Comment by slifty** [comment]
@Pamalosebi I don't htink @stanleyshly was asking a rhetorical question, but a genuine one.

They don't have enough information to know if the project is in a healthy, thriving state, which is why they asked what they asked!

---

@stanleyshly unfortunately right now the core maintainers of the project are a bit mental blocked on engaging with it.  A few community members offered to step into at least help lighten the community management load, but it's been several months and nobody's been empowered to do so.

New features aren't really being added, or merged in, and v5 is seriously lacking some very important things (e.g. support for profile pictures / files, as well as the ability to prevent new registrations).

All this is to say, it's not clear what the future of monica looks like, and it won't be clear until either the community decides:

1. it is time to attempt to fork the project and develop a separate branch (this is a big decision and in order to be successful would need intentionality and a solid core team with a common vision)

or

2. the original maintainers pick a few folks to empower in various ways to keep things moving.

My hope is still that 2 happens, though hope is all I can do!

**Comment by influjensbahr** [comment]
I only recently found this project and am using it quite a bit, it's a great help that can be super helpful with a few fixes and new features. It seems like there are a lot of people active here. What's stopping us from being active with the issues tracked (like tagging/commenting/helping) and submitting pull requests as fixes? Are pull requests currently not reviewed by anyone? 

What's the thing that's blocking this projects? It doesn't seem to be a lack of people wanting to help out.

**Comment by caesarshift** [comment]
As a new user (self-hosting), I'm also unsure of what to do. The docker project (I believe) uses v4, but now that I know that Chandler (v5) is the future, I'm hesitant to start using either v4 or v5 without a clear direction. In my opinion, the data migration shouldn't be a blocker. Finish MVP for v5 THEN worry about data migrations. The bigger issue is the pausing of all development on both branches. Paralysis of analysis is going to kill both versions.

**Comment by max-kibosh** [comment]
Putting in my two cents here, to reinforce public interest.
I'm in the same situation as @caesarshift - v4 is a great tool and helper, tried out v5 locally and it's such a great step in the right direction and exactly what I'm looking for, but obviously still lacking crucial functionality.

What some other teams have done (e.g. Gitea, HomeAssistant and others) is open a Discord server.
Imo, this can streamline communication, encourage interaction and provide great ways of getting the (scattered) community going.

I really don't want to see this project die, especially since v5 looks so promising!

**Comment by JoshuaBehrens** [comment]
Please do not move communication to a private (as in non searchable/indexable) gated community. This will make this place even less active. GitHub is already a collaboration tool with public communication like this.

**Comment by millette** [comment]
I'm with @JoshuaBehrens, please keep the discussions going here on GitHub. Discord is not an option for me and many people I know.

**Comment by lukasklein** [comment]
Not sure if this is the right place, but does anyone know what happened to the hosted version over at https://www.monicahq.com/? It's been down for me for the last two days

**Comment by Xyz00777** [comment]
@djaiss @asbiin what is the current state? if you dont have interest any more or have problems that would be absolutely okay, but its really sad that these awesome project is completely silently dying at the moment... :(

**Comment by noahkiss** [comment]
Is the unpinning of this issue our answer..?

**Comment by KaKi87** [comment]
Hello,
Does anyone have a docker-compose that just works ?
Thanks

**Comment by ctag** [comment]
This docker compose file is working for me with caddy-reverse-proxy.

```
# Run Monica with apache backend
services:
  app:
    image: monica:apache
    depends_on:
      - db
    ports:
      - 8200:80
    environment:
      - APP_ENV=production
      - DB_HOST=db
      - DB_DATABASE=monica
      - DB_USERNAME=monica
      - DB_PASSWORD=xxx
      - LOG_CHANNEL=stderr
      - CACHE_DRIVER=database
      - SESSION_DRIVER=database
      - QUEUE_DRIVER=sync
    volumes:
      - data:/var/www/html/storage
    restart: always
    networks:
      - caddy
      - default
    labels:
      caddy: domain.name
      caddy.reverse_proxy: "{{upstreams 80}}"
      caddy.import: fauth *
  db:
    image: mariadb:11
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=true
      - MYSQL_DATABASE=monica
      - MYSQL_USER=monica
      - MYSQL_PASSWORD=xxx
    volumes:
      - mysql:/var/lib/mysql
    restart: always
    networks:
      - default
volumes:
  data: {}
  mysql: {}
networks:
  default: {}
  caddy:
    external: true
```

**Comment by KaKi87** [comment]
I meant v5

**Comment by Szeraax** [comment]
If all you wanna do is try it out, you can just use this: 

```yaml
version: "3.3"
services:
  monica-next:
    ports:
      - 10080:80
    environment:
      - MAIL_MAILER=log
    image: ghcr.io/monicahq/monica-next:main
networks: {}
```

**Comment by KaKi87** [comment]
What about behing a reverse proxy on the host ? I'm using Apache with `ProxyPass` but the app still tries to make requests to `localhost`

**Comment by slifty** [comment]
I'd recommend that this thread isn't the right place for debugging docker compose: you may want to go to the discussions area and raise a question there @KaKi87

**Comment by KaKi87** [comment]
[I already did](https://github.com/monicahq/monica/discussions/7279), half a year ago, and tried mentioning the maintainers 12 days later, but never got a response.

**Comment by jncmney** [comment]
Is V5 still in active development? I was about to give it a go, but from what I’ve read it looks like it might have been abandoned.

**Comment by KaKi87** [comment]
The last non-bot commit is 3 years old, so I wouldn't say abandoned yet, but user support is definitely abandoned, as I haven't ever been able to build it and I never got help.

**Comment by slifty** [comment]
> Is V5 still in active development? I was about to give it a go, but from what I’ve read it looks like it might have been abandoned.

It had a long lapse but there have been PRs merged and new commits in the past month: https://github.com/monicahq/monica/commits/main/

**Comment by naiba** [comment]
Really hoping for an active development cadence, I really want to provide monica with access to other calendars, but since it's been unreleased and shows no signs of being active, it's hard for me to go ahead and start.

**Comment by KaKi87** [comment]
@djaiss What is [Rachel](https://github.com/monicahq/rachel) ? And, is Monica dead ?

**Comment by Pyromane** [comment]
Any Update? @djaiss @asbiin

**Comment by naiba** [comment]
First of all, I would like to thank the author for their contribution.
It's already 2026, and I believe GitHub has granted the authors free
Copilot access, but why is the project still stagnant?

**Comment by naiba** [comment]
I've spent a lot of time working on open-source project for free, and it now has 9k stars. I was just stating facts and expressing my confusion; I didn't attack the author. I'm not sure what the person who 👎 was thinking😂.

**Comment by naiba** [comment]
Okay guys, try AI, try Copilot, please.

**Comment by naiba** [comment]
Hey everyone 👋

Big fan of Monica and everything @djaiss and @asbiin have built over the years. This project has genuinely changed how many of us think about personal relationships.

That said, I noticed development has slowed down (understandably — side projects with full-time jobs are hard!), and Chandler has been in progress for a while. So I decided to do something about it.

I teamed up with an AI coding partner (Claude) and we **vibe-coded a complete ground-up rewrite of Monica in just a few sessions**:

👉 **[Bonds](https://github.com/naiba/bonds)** — a modern, fast, single-binary personal relationship manager.

### What's different?

| | Monica (PHP/Laravel) | Bonds (Go + React) |
|---|---|---|
| **Backend** | PHP / Laravel | Go (Echo + GORM) |
| **Frontend** | Vue (partial) | React 19 + TypeScript + Ant Design |
| **Database** | MySQL/PostgreSQL/SQLite | SQLite (zero config) |
| **Deployment** | PHP runtime + Composer + Node | Single binary, starts in milliseconds |
| **Tests** | — | 347 backend + 54 frontend + 6 E2E specs |
| **Docker** | Multi-service | Single container, arm64 + amd64 |

### Feature parity

Bonds already implements most of Monica's core features:

- Contacts, vaults, notes, tasks, reminders, gifts, debts, activities, life events, pets, groups, journals, goals, mood tracking
- **CardDAV / CalDAV** — sync with Apple Contacts, Thunderbird, etc.
- **Full-text search** (Bleve, CJK-aware)
- **vCard import/export**
- **2FA (TOTP)**, **WebAuthn/FIDO2**, **OAuth** (GitHub + Google)
- **Telegram notifications**, email reminders with cron
- File uploads, avatars, geocoding, audit log
- i18n (English + Chinese)

### How we built it

This was a **vibe coding** experiment — human direction + AI implementation. I provided the architecture decisions and feature requirements; Claude handled the bulk of the code generation, testing, and debugging. The result: a production-quality codebase with comprehensive test coverage, built in a fraction of the time it would normally take.

It proves that with the right human-AI collaboration, we can build secure, high-performance open-source software much more efficiently than before.

### Come build with us

Bonds is AGPL-3.0 (same as Monica) and fully open source. If you've been waiting for Monica to evolve, or if you want to contribute to a modern, lightweight alternative — PRs, issues, and ideas are all welcome:

🔗 https://github.com/naiba/bonds

Huge respect to the Monica team for the original vision. Bonds wouldn't exist without it. 🙏

**Comment by benjick** [comment]
@naiba v0.0.1. How long have you used it for yourself?

**Comment by noahkiss** [comment]
@naiba I'm sure the response here will be mixed, but I do think this space is ripe for new / forked projects such as yours, since the PRM open-source landscape is fairly sparse. Thanks for starting it off. Looking forward to a demo and/or screenshots

**Comment by KaKi87** [comment]
If a real project like Monica can't stay maintained then I wouldn't expect a vibe-coded alternative to do any better. I started with a wiki, Monica was already dead when I first heard of it, and I'm not gonna migrate to something Claude-maintained.

**Comment by Pyromane** [comment]
@naiba I think it needs a demo or at least screenshots to generate interest, as well as a migration option.
I will follow the development with interest.

**Comment by Na0mir** [comment]
Personally I'm following https://github.com/mattogodoy/nametag project, much more simple but does the job

**Comment by naiba** [comment]
Thanks everyone for the feedback! Replying here to avoid spamming notifications:

@benjick Just released recently, still in early dogfooding. There's now a live demo at https://bonds-dev.infiniteprogress.org/ — feel free to try it out. Bug reports and feedback are very welcome!

@noahkiss Thanks for the kind words! Demo is live (link above), feel free to give it a spin.

@Pyromane Demo at the link above, feel free to try it out. As for Monica data migration — I think the priority right now is for people to actually use it and find usability issues or bugs first. Migration only makes sense once the product itself is solid enough. Migration tooling will follow when the time is right.

@KaKi87 Understandable concern, let me share some context. I've been doing open source well before AI came along — years of experience that give me a clear vision for the architecture and direction of this project. With AI assistance, I rebuilt the entire thing in Go + React in roughly **7 hours** (while watching YouTube and vibe-coding, honestly). The low effort required to build and maintain is exactly why I can sustain this long-term. Claude is my implementer — every architecture decision, every code review, every aspect of maintaining this project is still me.

The codebase has 400+ tests with full CI. Code quality can be judged from the code itself, not from how it was written.

Curious — have you tried AI-assisted coding yourself? I'd genuinely recommend giving it a shot. I make careless mistakes too, and honestly AI is more rigorous than me most of the time. We use comprehensive unit tests and E2E tests to verify its work. The capability is already very strong.

**Comment by Pamalosebi** [comment]
What credentials do I need to log in to the demo?
Is there one with demo data?

**Comment by naiba** [comment]
> What credentials do I need to log in to the demo?
> Is there one with demo data?

My focus has been on refining and optimizing Bonds; I haven't created a demo account or set up data yet. A new account needs to be created for testing.

**Comment by b3nd1k** [comment]
@naiba I’ll be following this project with interest, the idea is great and I’m excited to see where it goes. I created a demo account but ran into several issues: photo uploads report success but never appear, logging calls, tasks, or reminders returns “The given data was invalid,” click "Edit" on a contact page does nothing, and adding a birthday fails with a 404 error. It feels like core functionality still needs some work, but I’m rooting for the project.

I’d also love to see PostgreSQL support. SQLite is fine for small datasets, but once you have thousands of contacts with notes, history, and reminders, Postgres offers much better performance, concurrency, and long-term scalability. OAuth2 support would be another big win, especially integration with providers like Authentik, as it would greatly improve self-hosting, SSO, and security for more advanced users.

**Comment by naiba** [comment]
@b3nd1k Please continue to pay attention. The issues you mentioned should have been created by others before. Although the AI has done a rough draft, there are still many minor problems. I am still working on some basic issues, and I will add support for common databases after it is improved.
